### PR TITLE
Refactor of ArrayHelper class

### PR DESCRIPTION
**Pull request by fcalderonnearsoft**
Made a refactor of `reverse` method. I made it a UnitTest and actually returned a reversed array.

Deleted a few methods that were commented and unused variables.

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/internal/util/collections/ArrayHelper.java [hunk 1]
```diff
@@ -428,13 +407,8 @@ public static int countNonNull(Serializable[] array) {
...
 
 	public static String[] reverse(String[] source) {
-		final int length = source.length;
-		final String[] destination = new String[length];
-		for ( int i = 0; i < length; i++ ) {
-			final int x = length - i - 1;
-			destination[x] = source[i];
-		}
-		return destination;
+		Collections.reverse( Arrays.asList( source ) );
```
Comment (vladmihalcea) [inline hibernate-core/src/main/java/org/hibernate/internal/util/collections/ArrayHelper.java]: If we change this, there will be a new `List` object created just for reversing the array. The Java version also created more garbage internally. This change also adds a side-effect. The original method creates a new array, and doesn't modify the source. This one will modify the source array.

Comment (vladmihalcea) [inline hibernate-core/src/main/java/org/hibernate/internal/util/collections/ArrayHelper.java]: I checked it and this method is not even used, so we might want to remove it anyway.

**Comment by vladmihalcea** [comment]
Actually, we could remove many more methods from this class. I asked the question on the mailing list to see if it's appropriate to do so. Thanks for the hint ;)

**Comment by fcalderonnearsoft** [comment]
Your welcome! Did you asked about a bunch of methods with deprecated annotation?

**Comment by vladmihalcea** [comment]
There are other methods that are not marked as deprecated and they can be replaces by Arrays.indexOf, for instance.

**Comment by fcalderonnearsoft** [comment]
Do you mean that we should delete `indexOf()` method and replace with `ArraysUtil.indexOf()` of Apache Commons Lang library?

**Comment by vladmihalcea** [comment]
One thing that's needed to be checked is whether those methods are not used by Hibernate Validator, OGM, Search as well.

**Comment by fcalderonnearsoft** [comment]
I undo changes in reverse method, I will wait until you check if methods are not used. 

Cheers.

**Comment by fcalderonnearsoft** [comment]
Hi @vladmihalcea did you checked if we could remove more methods?

**Comment by vladmihalcea** [comment]
I didn't have time to check it, sorry.

**Comment by vladmihalcea** [comment]
Applied upstream, thanks

**Pull request closed without merging**